src/components/Badge/IconOnlyBadge.js

```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Badge from './Badge';
+
+const IconOnlyBadge = ({ children, title, ...rest }) => (
+  <Badge title={title} {...rest}>
+    {children}
+    <span className="slds-assistive-text">{title}</span>
+  </Badge>
+);
+
+IconOnlyBadge.propTypes = {
+  /** The Icon to display. Can also be a string for notification-badges */
+  children: PropTypes.oneOfType([PropTypes.element, PropTypes.string]).isRequired,
+  /** Description of the Icon */
+  title: PropTypes.string.isRequired,
+};
+
+export default IconOnlyBadge;
```

Comment:
Good point. After creating this cmp, I noticed that Salesforce is inconsistent here and has a few badges in other component that need assistive-text and do not contain icons (despite the docs saying otherwise). Will change